java/code/src/com/redhat/rhn/manager/user/UserManager.java

```diff
@@ -148,6 +151,62 @@ public static boolean verifyPackagesAccess(Org org, List<Long> packageIds) {
         return (!dr.isEmpty());
     }
 
+    /**
+     * Get a concatenated stream of namespaces granted direct (to the user)
+     * or indirect (via access groups) access.
+     * @param user the user
+     * @return the stream of permitted namespaces
+     */
+    public static Stream<Namespace> getPermittedNamespaces(User user) {
+        return Stream.concat(user.getNamespaces().stream(),
```

Comment:
I think we can have this method with a more efficient implementation, where we could get all the user namespaces with a single query execution. We don't need to change it for this PR, but we should refactor it.

Reply:
I still need to test the performance of different approaches, but with eager fetching, Hibernate should perform good enough on this.